internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -341,6 +398,35 @@ func readPlainTextSecret(opts *CreateOpts) error {
 	return nil
 }
 
+func readConfigFile(opts *CreateOpts) (SecretConfig, error) {
+	var sc SecretConfig
+	f, err := os.ReadFile(opts.SecretFilePath)
+	if err != nil {
+		return sc, fmt.Errorf("unable to open config file: %w", err)
+	}
+
+	err = yaml.Unmarshal(f, &sc)
+	if err != nil {
+		return sc, fmt.Errorf("unable to unmarshal config file: %w", err)
+	}
+
+	return sc, nil
+}
+
+func validateFields(sc SecretConfig) []string {
+	var missingKeys []string
+
+	if sc.Type == "" || sc.IntegrationName == "" {
+		missingKeys = append(missingKeys, "type")
+	}
+
+	if sc.IntegrationName == "" {
+		missingKeys = append(missingKeys, "rotation_integration_name")
```

Comment:
Same here, I believe the validation would fail for dynamic secrets given it doesn't have a `rotation_integration_name` field?

Reply:
Sorted!